test/access/Ownable.test.js

```diff
@@ -1,72 +1,80 @@
-const { constants, expectEvent } = require('@openzeppelin/test-helpers');
-const { expectRevertCustomError } = require('../helpers/customError');
-
-const { ZERO_ADDRESS } = constants;
-
+const { ethers } = require('hardhat');
 const { expect } = require('chai');
-
-const Ownable = artifacts.require('$Ownable');
-
-contract('Ownable', function (accounts) {
-  const [owner, other] = accounts;
-
+const { loadFixture } = require('@nomicfoundation/hardhat-network-helpers');
+const { deploy, getFactory } = require('../helpers/deploy');
+
+async function fixture() {
+  const accounts = await ethers.getSigners(); // this is slow :/
+  const owner = accounts.shift();
+  const other = accounts.shift();
+  const ownable = await deploy('$Ownable', [owner.address]);
+  return { accounts, owner, other, ownable };
+}
+
+describe('Ownable', function () {
   beforeEach(async function () {
-    this.ownable = await Ownable.new(owner);
+    await loadFixture(fixture).then(results => Object.assign(this, results));
   });
 
   it('rejects zero address for initialOwner', async function () {
-    await expectRevertCustomError(Ownable.new(constants.ZERO_ADDRESS), 'OwnableInvalidOwner', [constants.ZERO_ADDRESS]);
+    // checking a custom error requires a contract, or at least the interface
+    // we can get it from the contract factory
+    const { interface } = await getFactory('$Ownable');
+
+    await expect(deploy('$Ownable', [ethers.ZeroAddress]))
+      .to.be.revertedWithCustomError({ interface }, 'OwnableInvalidOwner')
+      .withArgs(ethers.ZeroAddress);
   });
 
   it('has an owner', async function () {
-    expect(await this.ownable.owner()).to.equal(owner);
+    expect(await this.ownable.owner()).to.equal(this.owner.address);
   });
 
   describe('transfer ownership', function () {
     it('changes owner after transfer', async function () {
-      const receipt = await this.ownable.transferOwnership(other, { from: owner });
-      expectEvent(receipt, 'OwnershipTransferred');
+      await expect(this.ownable.connect(this.owner).transferOwnership(this.other.address))
+        .to.emit(this.ownable, 'OwnershipTransferred')
+        .withArgs(this.owner.address, this.other.address);
 
-      expect(await this.ownable.owner()).to.equal(other);
+      expect(await this.ownable.owner()).to.equal(this.other.address);
```

Comment:
I once wrote an helper that is
```
const getAddress = (account : string | { address: string }) : string => account.address ?? account.target ?? account;
```
but calling it everywhere is even worst.

Maybe it could be integrated by default ...

Reply:
`ethers.resolveAddress` already does it even better ...

Reply:
Note: I'd love to simplify the code by removing `.address` when its better supported, but I don't think we should wait for that. We should migrate with `.address` and then remove them when possible.